src/services/edusharing/logic/connector.js

```diff
@@ -211,25 +195,32 @@ class EduSharingConnector {
 		};
 
 		const parsed = await this.requestRepeater(options);
+		return this.EduSearchResponse(parsed);
+	}
 
-		// adds accesstoken to image-url to let user see the picture on client-side.
-		if (parsed && parsed.nodes) {
-			parsed.nodes.forEach((node) => {
-				if (node.preview && node.preview.url) {
-					node.preview.url += `&accessToken=${this.accessToken}`;
-				}
-			});
+	EduSearchResponse(parsed) {
+		if (!parsed) {
+			return {
+				total: 0,
+				limit: 0,
+				skip: 0,
+				data: [],
+			};
 		}
-
+		// filter out the resources without the external url
+		const filteredNodes = parsed.nodes.filter((node) => {
+			const location = node.properties['ccm:wwwurl'];
```

Comment:
I use ccm:wwwurl instead of cclom:location because cclom:location is available for each node even if it is restricted.

Reply:
please, solve this with Ioannis. this must be the external link which must be provided.

Reply:
Not necessary anymore